src/main/java/net/snowflake/ingest/streaming/internal/IcebergParquetValueParser.java

```diff
@@ -0,0 +1,440 @@
+    }
+
+    return new ParquetBufferValue(value, estimatedParquetSize);
+  }
+
+  /**
+   * Parses an int32 value based on Parquet logical type.
+   *
+   * @param value column value provided by user in a row
+   * @param type Parquet column type
+   * @param insertRowsCurrIndex Used for logging the row of index given in insertRows API
+   * @return parsed int32 value
+   */
+  private static int getInt32Value(
+      Object value, PrimitiveType type, final long insertRowsCurrIndex) {
+    LogicalTypeAnnotation logicalTypeAnnotation = type.getLogicalTypeAnnotation();
+    if (logicalTypeAnnotation == null) {
+      return DataValidationUtil.validateAndParseIcebergNumber(
+          type.getName(), value, insertRowsCurrIndex, Integer.class);
+    }
+    if (logicalTypeAnnotation instanceof DecimalLogicalTypeAnnotation) {
+      return getDecimalValue(value, type, insertRowsCurrIndex).unscaledValue().intValue();
+    }
+    if (logicalTypeAnnotation == LogicalTypeAnnotation.dateType()) {
+      return DataValidationUtil.validateAndParseDate(type.getName(), value, insertRowsCurrIndex);
+    }
+    throw new SFException(
+        ErrorCode.UNKNOWN_DATA_TYPE, logicalTypeAnnotation, type.getPrimitiveTypeName());
+  }
+
+  /**
+   * Parses an int64 value based on Parquet logical type.
+   *
+   * @param value column value provided by user in a row
+   * @param type Parquet column type
+   * @param insertRowsCurrIndex Used for logging the row of index given in insertRows API
+   * @return parsed int64 value
+   */
+  private static long getInt64Value(
+      Object value, PrimitiveType type, ZoneId defaultTimezone, final long insertRowsCurrIndex) {
+    LogicalTypeAnnotation logicalTypeAnnotation = type.getLogicalTypeAnnotation();
+    if (logicalTypeAnnotation == null) {
+      return DataValidationUtil.validateAndParseIcebergNumber(
+          type.getName(), value, insertRowsCurrIndex, Long.class);
+    }
+    if (logicalTypeAnnotation instanceof DecimalLogicalTypeAnnotation) {
+      return getDecimalValue(value, type, insertRowsCurrIndex).unscaledValue().longValue();
+    }
+    if (logicalTypeAnnotation instanceof TimeLogicalTypeAnnotation) {
+      return DataValidationUtil.validateAndParseTime(
+              type.getName(),
+              value,
+              timeUnitToScale(((TimeLogicalTypeAnnotation) logicalTypeAnnotation).getUnit()),
+              insertRowsCurrIndex)
+          .longValue();
+    }
+    if (logicalTypeAnnotation instanceof TimestampLogicalTypeAnnotation) {
+      boolean includeTimeZone =
+          ((TimestampLogicalTypeAnnotation) logicalTypeAnnotation).isAdjustedToUTC();
+      return DataValidationUtil.validateAndParseTimestamp(
+              type.getName(),
+              value,
+              timeUnitToScale(((TimestampLogicalTypeAnnotation) logicalTypeAnnotation).getUnit()),
+              defaultTimezone,
+              !includeTimeZone,
+              insertRowsCurrIndex)
+          .toBinary(false)
```

Comment:
we are removing timezone in all cases, is that correct behavior?

Reply:
According [here](https://github.com/snowflakedb/snowflake-ingest-java/blob/e808bbd6a215daaae9e5a49ec4006978afbeca8d/src/main/java/net/snowflake/ingest/streaming/internal/ParquetValueParser.java#L319C7-L324C30) and `Iceberg.timetamptz` is using `Snowflake.timestampltz` ([ref](https://github.com/snowflakedb/snowflake-ingest-java/blob/e808bbd6a215daaae9e5a49ec4006978afbeca8d/src/main/java/net/snowflake/ingest/streaming/internal/ParquetValueParser.java#L319C7-L324C30)).